os-server/src/main/java/app/openschool/faq/FaqServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package app.openschool.faq;
+
+import app.openschool.course.CourseRepository;
+import app.openschool.faq.api.dto.CreateFaqRequest;
+import app.openschool.faq.api.dto.UpdateFaqDtoRequest;
+import java.sql.SQLIntegrityConstraintViolationException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class FaqServiceImpl implements FaqService {
+
+  private final FaqRepository faqRepository;
+  private final CourseRepository courseRepository;
+
+  public FaqServiceImpl(FaqRepository faqRepository, CourseRepository courseRepository) {
+    this.faqRepository = faqRepository;
+    this.courseRepository = courseRepository;
+  }
+
+  @Override
+  public Page<Faq> findAll(Pageable pageable) {
+    return faqRepository.findAll(pageable);
+  }
+
+  @Override
+  @Transactional
+  public Page<Faq> findFaqsByCourseId(Long courseId, Pageable pageable) {
+    courseRepository.findById(courseId).orElseThrow(IllegalArgumentException::new);
+    return faqRepository.findFaqsByCourseId(courseId, pageable);
+  }
+
+  @Override
+  @Transactional
+  public Faq add(CreateFaqRequest request, String mentorEmail)
+      throws SQLIntegrityConstraintViolationException {
+    int updatedRows = faqRepository.saveFaq(request, mentorEmail);
```

Comment:
isn't it possible to return from repo the entity that just saved to not do one more read operation

Reply:
this option works for H2 DB but is not supported for MySql DB. In order not to create two repos depending on the profile, I propose this solution